src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +16,38 @@
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+            return actor;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot ADD actor from Database " + actor, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            Actor actor = session.get(Actor.class, id);
+            return Optional.of(actor);
```

Comment:
Consider using `Optional.ofNullable(actor)` instead of `Optional.of(actor)` to handle the case where `actor` might be `null`. This will prevent a `NullPointerException` if the actor with the given ID does not exist.